dask/dataframe/csv.py

```diff
@@ -0,0 +1,142 @@
+from __future__ import print_function, division, absolute_import
+
+from io import BytesIO
+from warnings import warn
+
+import numpy as np
+import pandas as pd
+
+from ..delayed import delayed
+from .io import from_delayed
+
+from ..bytes import read_bytes
+from ..bytes.compression import seekable_files, files as cfiles
+from ..utils import ensure_bytes
+
+
+delayed = delayed(pure=True)
+
+
+def bytes_read_csv(b, header, kwargs, dtypes=None, columns=None):
+    """ Convert a block of bytes to a Pandas DataFrame
+
+    Parameters
+    ----------
+    b: bytestring
+        The content to be parsed with pandas.read_csv
+    header: bytestring
+        An optional header to prepend to b
+    kwargs: dict
+        A dictionary of keyword arguments to be passed to pandas.read_csv
+    dtypes: dict
+        DTypes to assign to columns
+
+    See Also:
+        dask.dataframe.csv.read_csv_from_bytes
+    """
+    bio = BytesIO()
+    if not b.startswith(header.rstrip()):
+        bio.write(header)
+    bio.write(b)
+    bio.seek(0)
+    df = pd.read_csv(bio, **kwargs)
```

Comment:
Because pandas.read_csv is very complex.

Other keyword arguments like `parse_dates` screw with the column names.  Passing in dtypes of what you expect as an output does not achieve the desired effect.